src/components/User/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, useContext, useState } from 'react'
+import firebase from 'firebase'
+import { authentication } from '../../firebase'
+
+type UserType = {
+	user: firebase.User | null
+	isLoggedIn: boolean
+}
+
+export const userContext = createContext<UserType>({
+	user: null,
+	isLoggedIn: false,
+})
+
+const UserProvider: React.FC = ({ children }) => {
+	const [user, setUser] = useState(authentication.currentUser)
```

Comment:
When the type is more complicated, it's a good practice to type your state. :slightly_smiling_face:
```suggestion
	const [user, setUser] = useState<firebase.User | null>(authentication.currentUser)
```

Reply:
👍